tests/test_configspace_utils.py

```diff
@@ -0,0 +1,202 @@
+      "log": true,
+      "lower": 8,
+      "upper": 2980,
+      "default": 122
+    },
+    {
+      "name": "repl",
+      "type": "uniform_int",
+      "log": false,
+      "lower": 1,
+      "upper": 10,
+      "default": 6
+    },
+    {
+      "name": "max_depth",
+      "type": "uniform_int",
+      "log": false,
+      "lower": 3,
+      "upper": 10,
+      "default": 3
+    },
+    {
+      "name": "rate_drop",
+      "type": "uniform_float",
+      "log": false,
+      "lower": 0.0,
+      "upper": 1.0,
+      "default": 0.2
+    }
+  ],
+  "conditions": [
+    {
+      "child": "max_depth",
+      "parent": "booster",
+      "type": "IN",
+      "values": [
+        "dart",
+        "gbtree"
+      ]
+    },
+    {
+      "child": "rate_drop",
+      "parent": "booster",
+      "type": "EQ",
+      "value": "dart"
+    }
+  ],
+  "forbiddens": [],
+  "python_module_version": "0.4.18",
+  "json_format_version": 0.2
+}"""
+
+
+def test_parameterspace_from_configspace_with_conditions_and_log_transform():
+    cs_dict = json.loads(CS_CONDITIONS_JSON)
+    space = parameterspace_from_configspace_dict(cs_dict)
+    assert len(space) == 7
+    assert space.has_conditions()
+    assert space.check_validity(
+        {
+            "alpha": 1.0,
+            "booster": "gbtree",
+            "lambda": 1.0,
+            "nrounds": 122,
+            "repl": 6,
+            "max_depth": 3,
+        }
+    )
+    assert isinstance(
+        space._parameters["nrounds"]["parameter"]._transformation,
+        LogZeroOneIntegerTransformation,
+    )
+
+    _s = space.copy()
+    _s.fix(booster="dart")
+    assert len(_s.sample()) == 7
+
+    _s = space.copy()
+    _s.fix(booster="gblinear")
+    assert len(_s.sample()) == 5
+
+    alpha = cs_dict["hyperparameters"][0]
+    assert alpha["name"] == "alpha"
+    bounds = [alpha["lower"], alpha["upper"]]
+    assert list(space._parameters["alpha"]["parameter"].bounds) == bounds
+
+    booster = cs_dict["hyperparameters"][1]
+    assert booster["name"] == "booster"
+    assert space._parameters["booster"]["parameter"].values == booster["choices"]
+
+
+def test_parameterspace_from_configspace_with_normal_prior():
+    cs_dict = json.loads(
+        """{
+          "name": "myspace",
+          "hyperparameters": [
+            {
+              "name": "p",
+              "type": "normal_float",
+              "log": true,
+              "mu": 1.0,
+              "sigma": 10.0,
+              "default": 2.718281828459045
+            }
+          ],
+          "conditions": [],
+          "forbiddens": [],
+          "python_module_version": "0.6.0",
+          "json_format_version": 0.4
+        }
+        """
+    )
+    space = parameterspace_from_configspace_dict(cs_dict)
+    assert len(space) == 1
+
+    param = space.get_parameter_by_name("p")["parameter"]
+    assert isinstance(param._prior, TruncatedNormalPrior)
+    assert param._prior.mean == cs_dict["hyperparameters"][0]["mu"]
```

Comment:
I think a more meaningful test could be to draw samples from the parameterspace and compare the mean of those to the specified one (with some reasonable slack). That way, we would see if the TruncatedNormalPrior is initialized correctly above.